tools/lsp/preview/ui.rs

```diff
@@ -563,34 +571,181 @@ fn map_properties_to_ui(
         property_groups.push(property_group_from(&current_group, current_group_properties));
     }
 
-    Some(ElementInformation {
-        id: element.id.clone().into(),
-        type_name: element.type_name.clone().into(),
-        source_uri,
-        source_version,
-        range: to_ui_range(element.range)?,
+    Some((
+        ElementInformation {
+            id: element.id.clone().into(),
+            type_name: element.type_name.clone().into(),
+            source_uri,
+            source_version,
+            range: to_ui_range(element.range)?,
+        },
+        declarations,
+        Rc::new(VecModel::from(property_groups)).into(),
+    ))
+}
 
-        properties: Rc::new(VecModel::from(property_groups)).into(),
-    })
+enum Op {
+    Insert((usize, usize)),
+    Copy((usize, usize)),
+    PushBack(usize),
+    Remove(usize),
+}
+
+fn is_equal_value(c: &PropertyValue, n: &PropertyValue) -> bool {
+    c.value_bool == n.value_bool
+        && c.is_translatable == n.is_translatable
+        && c.kind == n.kind
+        && c.value_brush == n.value_brush
+        && c.value_float == n.value_float
+        && c.value_int == n.value_int
+        && c.default_selection == n.default_selection
+        && c.value_string == n.value_string
+        && c.tr_context == n.tr_context
+        && c.tr_plural == n.tr_plural
+        && c.tr_plural_expression == n.tr_plural_expression
+        && c.visual_items.row_count() == n.visual_items.row_count()
+        && std::iter::zip(c.visual_items.iter(), n.visual_items.iter()).all(|(c, n)| c == n)
+}
+
+fn is_equal(c: &PropertyInformation, n: &PropertyInformation) -> bool {
+    c.name == n.name && c.type_name == n.type_name && is_equal_value(&c.value, &n.value)
+}
+
+fn is_equal_element(c: &ElementInformation, n: &ElementInformation) -> bool {
```

Comment:
What's the difference with PartialEq ?

Reply:
Models are always different from each other :-( So *every* property is different from every other due to the embedded Models.